creator-node/test/audiusUsers.js

```diff
@@ -66,3 +85,91 @@ describe('test AudiusUsers', function () {
       .expect(200)
   })
 })
+
+// Below block uses actual ipfsClient (unlike first describe block), hence
+// another describe block for this purpose
+describe('tests /audius_users/metadata metadata upload with actual ipfsClient', function () {
+  let app, server, session, libsMock, ipfs
+
+  // Will need a '.' in front of storagePath to look at current dir
+  // a '/' will search the root dir
+  before(async () => {
+    let storagePath = config.get('storagePath')
+    if (storagePath.startsWith('/')) {
+      storagePath = '.' + storagePath
+      config.set('storagePath', storagePath)
+    }
+  })
+
+  beforeEach(async () => {
+    ipfs = ipfsClient.ipfs
+    libsMock = getLibsMock()
+
+    const appInfo = await getApp(ipfs, libsMock, BlacklistManager)
+    await BlacklistManager.blacklist(ipfs)
+
+    app = appInfo.app
+    server = appInfo.server
+    session = await createStarterCNodeUser()
+  })
+
+  afterEach(async () => {
+    sinon.restore()
+    await server.close()
+  })
+
+  it('should fail if metadatda is not found in request body', async function () {
+    const resp = await request(app)
+      .post('/audius_users/metadata')
+      .set('X-Session-ID', session)
+      .send({ dummy: 'data' })
+      .expect(500)
+
+    assert.deepStrictEqual(resp.body.error, 'Internal server error')
+  })
+
+  it('should throw error response if saving metadata to fails', async function () {
+    sinon.stub(ipfs, 'add').rejects(new Error('ipfs add failed!'))
+
+    const metadata = { metadata: 'spaghetti' }
+    const resp = await request(app)
+      .post('/audius_users/metadata')
+      .set('X-Session-ID', session)
+      .send(metadata)
+      .expect(500)
+
+    assert.deepStrictEqual(resp.body.error, 'Could not save file to disk, ipfs, and/or db: Error: ipfs add failed!')
+  })
+
+  it('should successfully adds metadata file to filesystem, db, and ipfs', async function () {
+    const metadata = sortKeys({ spaghetti: 'spaghetti' })
+    const resp = await request(app)
+      .post('/audius_users/metadata')
+      .set('X-Session-ID', session)
+      .send({ metadata })
+      .expect(function (res) {
```

Comment:
what's purpose of this? just to capture any unexpected error msgs?

Reply:
yup! so if any unexpected errors occur, the test would fail as we expect this request to respond with a 200

Reply:
yeah that's a good point, removing this expect check!